### ISSUE TITLE
Undocumented modes on the VT340?

### ISSUE THREAD

**Issue by j4james** [hackerb9/vt340test#27]
I was just looking at your `checksixelmode.sh` script, and I was wondering whether you had ever tried running it in a loop on your VT340, like from 1 to 200, just to see what comes up? I'm curious whether there are any modes that it supports that aren't actually documented anywhere.

The reason I ask, is because the DEC STD-070 manual has a reference to a _Graphics Cursor Enable Mode_ (`DECGCEM`), although it never lists a mode number for that. I figured if any terminal was likely to support that mode it would be the VT340, and we might be able to find it with a `DECRQM` scan.

Also, I noticed a couple of modes that are missing from your list of descriptions, in case you want to add to that. They're mostly from the VT382 models.

```
Ps[9]="Interlace mode (DECINLM), VT100"
Ps[12]="Katakana Shift Mode (DECKANAM), VT382-J"
Ps[49]="Thai Input Mode (DECTHAIM), VT382-T"
Ps[50]="Thai Cursor Mode (DECTHAICM), VT382-T"
Ps[59]="Kanji/Katakana Display Mode (DECKKDM), VT382-J"
Ps[90]="Thai Space Compensating Mode (DECTHAISCM), VT382-T"
```

Then I've got a few more from the DEC VK100, but they're a bit weird. They don't give the typical `DECxxxM` name anywhere, but just list the two character ID that is used when configuring the mode from the setup. I'll leave it to you to decide whether you think they're worth including.

```
Ps[20]="Overstrike Mode (OS), VK100"
Ps[21]="Local BASIC Mode (BA), VK100"
Ps[22]="Host BASIC Mode (BA), VK100"
Ps[23]="Programmed Keypad Mode (PK), VK100"
Ps[24]="Auto Hardcopy Mode (AH), VK100"
```

**Comment by hackerb9** [comment]
I've never tried it in a loop, but that's a good idea. 

I believe the VT340's graphics cursor can be enabled in Tektronix mode and, IIRC, in ReGIS. I could see it also being a DEC private mode.  

I've added the mode names you found. My script doesn't yet show two different mode names (e.g., "Interlace Mode" and "X10 Mouse Mode" for mode 9), so I have to pick one. Typically, I've been going with whatever is better documented and seems more "official", but it looks like for mode 9, I should go with "mouse mode" as "interlace" seems to have been dropped for VT2xx and up. If you have thoughts about how such collisions should be best handled, I'm open to suggestions.

**Comment by j4james** [comment]
> I believe the VT340's graphics cursor can be enabled in Tektronix mode and, IIRC, in ReGIS.

That was my understanding too, but I've never seen them refer to ReGIS and Tektronix operations with the `DECxxx` naming scheme, which is why I was thinking it might also have been exposed as a regular DEC escape sequence too.

> Typically, I've been going with whatever is better documented and seems more "official", but it looks like for mode 9, I should go with "mouse mode" as "interlace" seems to have been dropped for VT2xx and up.

I'm personally in favor of giving preference to DEC over anything else, just because I'm more interested in DEC terminal emulation than I am in modern terminals. But ignoring my bias, this probably does make the most sense. As you say, `DECINLM` was limited to the VT1xx devices, so it's not like it applies to the VT340.

Mode 12 is a little harder to decide. I suspect the AT&T 610 blink mode is much better known than `DECKANAM`, and certainly more widely implemented in modern terminals. But it also feels like the VT382 is in the same family as the VT340, so perhaps that's an argument for giving it priority.

The other edge case is mode 35, but I don't think the Rxvt font mode is really any more widely known than `DECHEBM`, so giving priority to DEC here seems reasonable.

The remaining conflicts I think are all with VT340 modes, so I'd say you've obviously got to give DEC the priority there.

**Comment by j4james** [comment]
> I see no evidence of DECGCEM

Oh well, it was worth a shot. But thanks for doing the scan. There are still some interesting things in that report. Like it never occurred to me that `DECNRCM` would be permanently reset. I'm assuming that's because you've got it configured to use a US keyboard. But it's somewhat surprising then that `DECKBUM` isn't also permanently reset.

**Comment by hackerb9** [comment]
I wondered if DECNRCM was something I had configured when I was adjusted my VT340 to use Latin-1, so I reset the settings and ran the tests again. That gives only [slightly different results](https://github.com/hackerb9/vt340test/blob/main/j4james/trawlmodes-default.md) and DECNRCM is still permanently reset. 

According to the VT340 Text Programming manual:

> The terminal ignores DECNRCM if you set the Keyboard Dialect
> feature in Keyboard Set-Up screen to "North American”. If you
> set the dialect to “British” the terminal recognizes DECNRCM.

My VT340-+ (AKA VT340-G2) defaults to the North American keyboard dialect, but I assume that's just due to the market it was sold in and others would have different defaults. I changed the keyboard to a random nationality (Flemish) and sure enough, DECNRCM now shows up as "reset" instead of "permanently reset". I was able to set and reset it as one would expect.
